eyeshade/controllers/snapshots.js

```diff
@@ -0,0 +1,146 @@
+  updatedAt: updatedAtValidator,
+  items: accountBalancesListValidator
+})
+
+const upsertBalanceSnapshot = `
+insert into balance_snapshots (id)
+values ($1)
+returning *;
+`
+const getOneSnapshotBalance = `
+select
+  id,
+  completed,
+  created_at as "createdAt",
+  updated_at as "updatedAt"
+from balance_snapshots
+where
+  id = $1
+limit 1
+`
+const getSnapshotBalanceAccounts = `
+select
+  account_id as "accountId",
+  account_type as "accountType",
+  balance
+from
+  balance_snapshot_accounts
+where
+  snapshot_id = $1
+`
+
+v1.createSnapshot = {
+  handler: createSnapshotHandler,
+  auth: {
+    strategy: 'simple-scoped-token',
+    scope: ['publishers'],
+    mode: 'required'
+  },
+  description: 'Used by antifraud service for generating payload snapshots',
+  tags: ['api', 'publishers'],
+  validate: {
+    payload: Joi.object().keys({
+      snapshotId: snapshotIdValidator
+    }).unknown(true)
+  },
+  response: {
+    schema: Joi.object().keys({
+      snapshotId: snapshotIdValidator
+    })
+  }
+}
+
+v1.getFullSnapshot = {
+  handler: getFullSnapshotHandler,
+  auth: {
+    strategy: 'simple-scoped-token',
+    scope: ['publishers', 'antifraud'],
+    mode: 'required'
+  },
+  description: 'Used by antifraud service for generating payload snapshots',
+  tags: ['api', 'publishers', 'antifraud'],
+  validate: {
+    params: Joi.object().keys({
+      snapshotId: snapshotIdValidator
+    }).unknown(true)
+  },
+  response: {
+    schema: fullSnapshotValidator
+  }
+}
+
+module.exports.routes = [
+  braveHapi.routes.async().post().path('/v1/snapshots/').whitelist().config(v1.createSnapshot),
+  braveHapi.routes.async().path('/v1/snapshots/{snapshotId}').whitelist().config(v1.getFullSnapshot)
+]
+
+function createSnapshotHandler (runtime) {
+  return async (request, h) => {
+    const debug = braveHapi.debug(module, request)
+    const { snapshotId } = request.payload
+    try {
+      await runtime.postgres.query(upsertBalanceSnapshot, [snapshotId])
+    } catch (e) {
+      throw braveUtils.postgresToBoom(e)
+    }
+    await runtime.queue.send(debug, 'update-snapshot-accounts', {
+      snapshotId
+    })
+    return {
+      snapshotId
+    }
+  }
+}
+
+function getFullSnapshotHandler (runtime) {
+  return async (request, h) => {
+    const { snapshotId } = request.params
+    const roClient = await runtime.postgres.roConnect()
+    // this can probably done in one query
+    const { rows: snapshots, rowCount } = await roClient.query(getOneSnapshotBalance, [snapshotId])
+    if (!rowCount) {
+      roClient.release()
+      throw boom.notFound()
+    }
+    const snapshot = snapshots[0]
+    snapshot.items = []
+    if (!snapshot.completed) {
+      roClient.release()
+      return h.response(snapshot).code(202)
+    }
+    const { rows: accountBalances } = await roClient.query(getSnapshotBalanceAccounts, [snapshotId])
+    snapshot.items = accountBalances
+    roClient.release()
+    return snapshot
+  }
+}
```

Comment:
I think what we want from the GET is to expose the ability for publishers to get the balance of a specific account_id within a particular snapshot, rather than bulk GET endpoint to get the entire snapshot.

This basically would be used on the Publishers admin dashboard so admins could go to a particular publisher's page and see 'oh this account is supposed to receive 30 BAT this settlement'